nereid/nereid/api/api_v1/endpoints/treatment_facilities.py

```diff
@@ -123,7 +121,9 @@ async def initialize_treatment_facility_parameters(
     treatment_facilities, context = tmnt_facility_req
 
     task = bg.background_initialize_treatment_facilities.s(
-        treatment_facilities=treatment_facilities.dict(), context=context
+        treatment_facilities=treatment_facilities.dict(),
+        pre_validated=True,
```

Comment:
checkout the function args to the `initialize_treatment_facility_parameters` function. in fastapi we can inject functions that catch the data that gets posted to our routes by using the `fastapi.Depends` function. in this case, i'm injecting the result of a validator function defined right before this function, called `validate_facility_request`. This intercepts all the args that are posted to the endpoint (the treatment_facility object and the context, which also has a separate dependency) and runs the `validate_treatment_facility_models` script on the unvalidated data before passing it to the endpoint.

all of this is to ensure that before we pass bad data to the background process (where exceptions register to the client as 'internal server errors' and are utterly useless) we perform validation on the submission to identify errors and return a json object detailing the error, rather than just a 500 error code.